### ISSUE TITLE
The CouchDB Helm chart has an issue

### ISSUE THREAD

**Issue by Noksa** [apache/couchdb-helm#55]
**Describe the bug**
The CouchDB helm chart has an issue with annotations key.
You can find the error in this line: https://github.com/apache/couchdb-helm/blob/main/couchdb/templates/statefulset.yaml#L23

**Version of Helm and Kubernetes**:

**What happened**:
An error occurs if we use the top `annotations` key in a custom values file:
```
  annotations:
    sidecar.istio.io/inject: "false"
```
The error:
```
Error: UPGRADE FAILED: template: smb-cluster/charts/modb/templates/statefulset.yaml:23:28: executing "smb-cluster/charts/modb/templates/statefulset.yaml" at <include (print $.Template.BasePath "/configmap.yaml") .>: error calling include: template: smb-cluster/charts/db/templates/_helpers.tpl:14:14: executing "couchdb.fullname" at <.Values.fullnameOverride>: nil pointer evaluating interface {}.fullnameOverride
```
**What you expected to happen**:
We expect it works...

**How to reproduce it** (as minimally and precisely as possible):

**Anything else we need to know**:
You should replace "." with "$" when you pass the parameter to include func, because you use `with` block before and "." inside this block is not the root context.

So instead of this:
```
checksum/config: {{ include (print $.Template.BasePath "/configmap.yaml") . | sha256sum }}
```
It should be:
```
checksum/config: {{ include (print $.Template.BasePath "/configmap.yaml") $ | sha256sum }}
```

**Comment by bnhoogvliet** [comment]
We are experiencing the same issue when setting annotations using the `annotations` key. Any update on this issue?

**Comment by kocolosk** [comment]
Ah! @willholley actually addressed this with a slightly different fix in #38, but we never ended up merging it. I'll take a look at rebasing it today.
